### PR TITLE
small corrections in the docs

### PR DESCRIPTION
**Pull request by jalogisch**
just some small corrections for the documentation - changed versions and removed some warning that are not longer neede.

additional added the 2.3 upgrade nodes from the server repo

### CODE REVIEW

File: pages/installation.rst [hunk 1]
```diff
@@ -42,4 +42,4 @@ The Graylog server application has the following prerequisites:
 * `MongoDB 2.4 or later <https://docs.mongodb.org/manual/administration/install-on-linux/>`_ (latest stable version is recommended)
 * Oracle Java SE 8 or later (OpenJDK 8 also works; latest stable update is recommended)
 
-.. caution:: Graylog 2.x **does not** work with Elasticsearch 5.x!
+.. caution:: Graylog prior 2.3 **does not** work with Elasticsearch 5.x!
```
Comment (joschi) [inline pages/installation.rst]:
```text
Graylog versions prior to 2.3.0 **do not** work with Elasticsearch 5.x
```

File: pages/securing.rst [hunk 2]
```diff
@@ -24,13 +24,13 @@ All parts of one Graylog installation will communicate over network sockets. Dep
     * - Graylog (web interface / API)
       - 9000 (tcp)
     * - Graylog to Elasticsearch
-      - 9350 (tcp)
+      - 9200 (tcp)
```
Comment (joschi) [inline pages/securing.rst:27]: 👍

File: pages/upgrade/graylog-2.3.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
+**************************
+Upgrading to Graylog 2.3.x
+**************************
+
+.. _upgrade-from-22-to-23:
+
+This file only contains the upgrade note for the upcoming release.
+Please see `our documentation <http://docs.graylog.org/en/latest/pages/upgrade.html>`_
```
Comment (joschi) [inline pages/upgrade/graylog-2.3.rst]: This link could be replace by an internal reference.

File: pages/upgrade/graylog-2.3.rst [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+for the complete upgrade notes.
+
+Graylog switches to Elasticsearch HTTP client
+=============================================
+
+In all prior versions, Graylog used the Elasticsearch node client to connect to an Elasticsearch cluster, which was acting as a client-only Elasticsearch node. For compatibility reasons of the used binary transfer protocol, the range of Elasticsearch versions Graylog could connect to was limited. For more information and differences between the different ways to connect to Elasticsearch, you can check the `Elasticsearch documentation <https://www.elastic.co/guide/en/elasticsearch/guide/current/_talking_to_elasticsearch.html>`_.
```
Comment (joschi) [inline pages/upgrade/graylog-2.3.rst]: The URL should point to a specific version: https://www.elastic.co/guide/en/elasticsearch/guide/5.4/_talking_to_elasticsearch.html

**Review by joschi — CHANGES_REQUESTED** [review]

**Review by joschi — APPROVED** [review]

File: pages/upgrade/graylog-2.3.rst [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
++----------------------------------------------------+-----------+--------------------------------------------------------------+-----------------------------+
+| ``elasticsearch_discovery_filter``                 | String    | Filter by node attributes for the discovered nodes           | empty (use all nodes)       |
++----------------------------------------------------+-----------+--------------------------------------------------------------+-----------------------------+
+| ``elasticsearch_discovery_frequency``              | Duration  | Frequency of the Elasticsearch node discovery                | ``30s`` (30 Seconds)        |
++----------------------------------------------------+-----------+--------------------------------------------------------------+-----------------------------+
+
+In most cases, the only configuration setting that needs to be set explicitly is ``elasticsearch_hosts``. All other configuration settings should be tweaked only in case of errors.
+
+.. warn:: The automatic node discovery does not work if Elasticsearch requires authentication, e. g. when using Shield (X-Pack).
```
Comment (joschi) [inline pages/upgrade/graylog-2.3.rst]: The correct admonition is "warning": http://docutils.sourceforge.net/docs/ref/rst/directives.html#warning

**Review by joschi — CHANGES_REQUESTED** [review]

File: pages/upgrade/graylog-2.3.rst (hunk 5 above)
Comment (jalogisch) [reply pages/upgrade/graylog-2.3.rst]: opas missed that. Just copied it over from the server repo.

will change

**Review by joschi — APPROVED** [review]

**Pull request merged**